src/ray/core_worker/gcs_server_address_updater.cc

```diff
@@ -20,39 +20,39 @@ namespace core {
 GcsServerAddressUpdater::GcsServerAddressUpdater(
     const std::string raylet_ip_address, const int port,
     std::function<void(std::string, int)> update_func)
-    : update_func_(update_func) {
-  // Init updater thread and run its io service.
-  updater_thread_.reset(new std::thread([this] {
-    SetThreadName("gcs_address_updater");
-    /// The asio work to keep io_service_ alive.
-    boost::asio::io_service::work io_service_work_(updater_io_service_);
-    updater_io_service_.run();
-  }));
-  client_call_manager_.reset(new rpc::ClientCallManager(updater_io_service_));
-  auto grpc_client =
-      rpc::NodeManagerWorkerClient::make(raylet_ip_address, port, *client_call_manager_);
-  raylet_client_ = std::make_shared<raylet::RayletClient>(grpc_client);
-  // Init updater runner.
-  updater_runner_.reset(new PeriodicalRunner(updater_io_service_));
+    : client_call_manager_(updater_io_service_),
+      raylet_client_(rpc::NodeManagerWorkerClient::make(raylet_ip_address, port,
+                                                        client_call_manager_)),
+      update_func_(update_func),
+      updater_runner_(updater_io_service_),
+      updater_thread_([this] {
+        SetThreadName("gcs_address_updater");
+        std::thread::id this_id = std::this_thread::get_id();
+        RAY_LOG(INFO) << "GCS Server updater thread id: " << this_id;
+        /// The asio work to keep io_service_ alive.
+        boost::asio::io_service::work io_service_work_(updater_io_service_);
+        updater_io_service_.run();
+      }) {
   // Start updating gcs server address.
-  updater_runner_->RunFnPeriodically(
+  updater_runner_.RunFnPeriodically(
       [this] { UpdateGcsServerAddress(); },
-      RayConfig::instance().gcs_service_address_check_interval_milliseconds());
+      RayConfig::instance().gcs_service_address_check_interval_milliseconds(), "ABCDE");
```

Comment:
Using a meaningful name here? Ok to fix this in a subsequent PR.

Reply:
It was a mistake haha.. thanks for catching it!